test/Servant/API/ContentTypes/AvroSpec.hs

```diff
@@ -1,10 +1,78 @@
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeOperators     #-}
+
 module Servant.API.ContentTypes.AvroSpec (spec) where
 
+import           Control.Concurrent            (forkIO)
+import           Control.Monad                 (void)
+import           Control.Monad.IO.Class        (liftIO)
+import           Data.Proxy                    (Proxy (..))
+import           Data.Text                     (Text, pack)
+import           Network.HTTP.Client           (defaultManagerSettings,
+                                                newManager)
+import qualified Network.Wai.Handler.Warp      as Warp
+import           Servant.API
+import           Servant.API.ContentTypes.Avro
+import           Servant.Client
+import           Servant.Server
+
 import           Test.Hspec
+import           Test.QuickCheck
+import           Test.QuickCheck.Monadic
+
+
+
+newtype ArbitraryText = ArbitraryText Text deriving Show
+
+instance Arbitrary ArbitraryText where
+  arbitrary = ArbitraryText . pack <$> arbitrary
+
+
+
+type TestAPI =
+  "put" :> ReqBody '[Avro, JSON] Text :> Put '[JSON] Text
+  :<|>
+  "get" :> Get '[Avro] (Maybe Text)
+
+testApp :: Application
+testApp = serve (Proxy :: Proxy TestAPI) handlers
+  where
+    handlers = handlePut :<|> handleGet
+
+    handlePut :: Text -> Handler Text
+    handlePut input = return $ input <> "x"
+
+    handleGet :: Handler (Maybe Text)
+    handleGet = return (Just "weewoo")
+
+-- The app is stateless so we don't need a fresh server each time we run a test.
+startTestApp :: IO ()
+startTestApp = void . liftIO . forkIO $ Warp.run 8888 testApp
```

Comment:
More of an idle comment: 

I would have assumed that not killing the forked thread would prevent the test suite from terminating, but Travis looks like it's a-okay with this so I guess cleanup Just Works.

Reply:
Yeah, it looks like a `forkIO`ed thread is what gets called a daemon thread in other languages. The behavior's the same on OS X.